src/polygon/src/polygon.cpp

```diff
@@ -536,8 +542,13 @@ class Monotones {
           PRINT("End");
           CloseEnd(vert);
           return End;
-        } else if (westPair != activePairs_.end() &&
-                   std::next(westPair) == eastPair) {
+        } else if (!eastPair->westCertain || !westPair->eastCertain ||
+                   (westPair != activePairs_.end() &&
+                    std::next(westPair) == eastPair)) {
+          if (!eastPair->westCertain)
+            activePairs_.splice(std::next(westPair), activePairs_, eastPair);
+          if (!westPair->eastCertain)
+            activePairs_.splice(eastPair, activePairs_, westPair);
```

Comment:
This is the key change, reordering uncertain pairs based on how they get attached to each other.